### ISSUE TITLE
toPreProcess NullPointerException when using MinMaxNormalizer

### ISSUE THREAD

**Issue by vladtc** [deeplearning4j/deeplearning4j#3846]
#### Issue Description

Expected behavior: I was expecting to have no errors, since the equivalent classes used in 0.8.0 work just fine.

Encountered behavior: I got the following error after all the training epochs for the first neural network candidate produced by the CandidateGenerator class were finished:

Exception in thread "ADSI prefetch thread" java.lang.RuntimeException: java.lang.NullPointerException: toPreProcess	at org.deeplearning4j.datasets.iterator.AsyncDataSetIterator$AsyncPrefetchThread.run(AsyncDataSetIterator.java:442)
Caused by: java.lang.NullPointerException: toPreProcess	at org.nd4j.linalg.dataset.api.preprocessor.AbstractDataSetNormalizer.preProcess(AbstractDataSetNormalizer.java:113)	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:295) at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:393)	at org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator.next(RecordReaderDataSetIterator.java:51) at org.deeplearning4j.datasets.iterator.AsyncDataSetIterator$AsyncPrefetchThread.run(AsyncDataSetIterator.java:417)

The code producing this is mentioned in https://gist.github.com/vladtc/f854164ce05d887ee377df3964de5251
and https://gist.github.com/vladtc/7ffefec6eca2b56c42305c47c5d0d43d

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.9.1-SNAPSHOT and 0.9.1
* platform information (OS, etc): 16.04.2-Ubuntu
* CUDA version, if used: no GPU used but cuda 7.5 and 8.0 were present in the arbiter pom
* NVIDIA driver version, if in use: no driver used

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

**Comment by vladtc** [comment]
Attached are the 4 data files needed to reproduce this. Extension needs to be changed from .txt to .csv

[smth_test.txt](https://github.com/deeplearning4j/deeplearning4j/files/1218385/smth_test.txt)
[smth_train.txt](https://github.com/deeplearning4j/deeplearning4j/files/1218384/smth_train.txt)
[smth_validate.txt](https://github.com/deeplearning4j/deeplearning4j/files/1218386/smth_validate.txt)
[smth_whole.txt](https://github.com/deeplearning4j/deeplearning4j/files/1218388/smth_whole.txt)

**Comment by vladtc** [comment]
Got the same error using the recently released 0.9.1 version.

**Comment by vladtc** [comment]
Attached are the normalized data files that can be used to execute the program without using the normalizer. The error encountered is: 
[main] INFO org.deeplearning4j.arbiter.optimize.runner.BaseOptimizationRunner - Task [i] failed during execution

Note: trainFile/validateFile/testFile/wholeFile variables need to be initialized to "smth_train_normalized.txt", "smth_validate_normalized.txt" etc.

[smth_test_normalized.txt](https://github.com/deeplearning4j/deeplearning4j/files/1225040/smth_test_normalized.txt)
[smth_train_normalized.txt](https://github.com/deeplearning4j/deeplearning4j/files/1225042/smth_train_normalized.txt)
[smth_validate_normalized.txt](https://github.com/deeplearning4j/deeplearning4j/files/1225041/smth_validate_normalized.txt)
[smth_whole_normalized.txt](https://github.com/deeplearning4j/deeplearning4j/files/1225043/smth_whole_normalized.txt)

**Comment by atradeus** [comment]
This is a bug in the RecordReaderDataSetIterator class.  Specifically, you need to set last = ds in the mdsToDataSet function.  If I get a chance, I'll create a patch, but for now we can work around the bug.
```    
    private DataSet mdsToDataSet(MultiDataSet mds) {

       // code cut for brevity

        if (preProcessor != null) {
            preProcessor.preProcess(ds);
        }

        last = ds;

        return ds;
    }

**Comment by AlexDBlack** [comment]
https://github.com/deeplearning4j/deeplearning4j/pull/4326
